modules/nextflow/src/main/groovy/nextflow/executor/LsfExecutor.groovy

```diff
@@ -35,6 +35,37 @@ class LsfExecutor extends AbstractGridExecutor {
 
     private static char BLANK = ' ' as char
 
+    /**
+    * parse the memory units from $LSF_ENVDIR/lsf.conf, 
+    * which should be accessible via all nodes. Otherwise, default is KB
+    * -- see https://www.ibm.com/support/knowledgecenter/en/SSETD4_9.1.3/lsf_config_ref/lsf.conf.lsf_unit_for_limits.5.html
+    */
+    def getLSFmemoryUnits() {
+        // check environment variable exists
+        def envDir = System.getenv("LSF_ENVDIR") 
+        if ( !envDir ) return null
+        def envFile = new File(envDir, "lsf.conf")
+        if ( !envFile.exists() ) return null
+        for (def line : envFile.readLines() ){
+            if ( line.startsWith("LSF_UNIT_FOR_LIMITS=") ){
+                def memoryUnits = line.split("LSF_UNIT_FOR_LIMITS=")[1]
+            }
+        }
+        return(memoryUnits)
+    }
+
+
+    /**
+     * Add an init method in the LsfExecutor overriding the base init() in AbstractGridExecutor.groovy
+    */
+    @Override
+    void init() {
+        super.init()
+        queueInterval = session.getQueueStatInterval(name)
+        log.debug "Creating executor '$name' > queue-stat-interval: ${queueInterval}"
+        def memoryUnitLSF = getLSFmemoryUnits()
+    }
```

Comment:
`memoryUnitLSF` has to be a class attributed otherwise it can't be accessed outside `init`

Reply:
Also it would make sense to default to `MB` when `getLSFMem` is returning `null` e.g. 

```
memoryUnitLSF = getLSFmemoryUnits() ?: 'MB'
```

Reply:
> Also it would make sense to default to `MB` when `getLSFMem` is returning `null` e.g.
> 
> ```
> memoryUnitLSF = getLSFmemoryUnits() ?: 'MB'
> ```

Yes, this seems very reasonable. This would also negate the requirement for an `if` statement below. 

Given the manual states the default is `KB`, I suppose we should use `KB`.

Reply:
But the would alter the current assumption of NF that are megabytes.

Reply:
> But the would alter the current assumption of NF that are megabytes.

Ah, this is a NF-wide assumption?

Reply:
I will open a separate issue regarding, but this PR should not change the current behaviour if not mem unit is specified. Do you have a link LSF documentation as reference ?